src/main/java/com/hubspot/jinjava/lib/filter/TruncateHtmlFilter.java

```diff
@@ -64,87 +57,21 @@ public String getName() {
   public Object filter(
     Object var,
     JinjavaInterpreter interpreter,
-    Object[] args,
-    Map<String, Object> kwargs
+    Map<String, Object> parsedArgs
   ) {
-    String length = null;
-    if (kwargs.containsKey(LENGTH_KEY)) {
-      length = Objects.toString(kwargs.get(LENGTH_KEY));
-    }
-    String end = null;
-    if (kwargs.containsKey(END_KEY)) {
-      end = Objects.toString(kwargs.get(END_KEY));
-    }
-    String breakword = null;
-    if (kwargs.containsKey(BREAKWORD_KEY)) {
-      breakword = Objects.toString(kwargs.get(BREAKWORD_KEY));
-    }
-
-    String[] newArgs = new String[3];
-    for (int i = 0; i < args.length; i++) {
-      if (i >= newArgs.length) {
-        break;
-      }
-      newArgs[i] = Objects.toString(args[i]);
-    }
-
-    if (length != null) {
-      newArgs[0] = length;
-    }
-    if (end != null) {
-      newArgs[1] = end;
-    }
-    if (breakword != null) {
-      newArgs[2] = breakword;
-    }
-
-    if (var instanceof SafeString) {
-      return filter((SafeString) var, interpreter, newArgs);
-    }
-
-    return filter(var, interpreter, newArgs);
-  }
-
-  @Override
-  public Object filter(Object var, JinjavaInterpreter interpreter, String... args) {
-    if (var instanceof String) {
-      int length = DEFAULT_TRUNCATE_LENGTH;
-      String ends = DEFAULT_END;
-
-      if (args.length > 0) {
-        try {
-          length = Integer.parseInt(Objects.toString(args[0]));
-        } catch (Exception e) {
-          ENGINE_LOG.warn(
```

Comment:
wow, this removes a lot of code! Can you ensure that this is still covered in the unit tests?

Reply:
Ah I see your point, it ignores parsing errors. Given that we are using `defaultValue` now did you want to keep `parse-and-ignore-error` functionality or rely on correct user input/sensible default value?

Reply:
I think we should just confirm that it still validates the value but we don't need to log a warning. It's pretty useless.

Reply:
Take a look at the last commits, I've added a default on error. Additionally I've fixed up the number parsing as NumberUtils seems to default to 0 if not parsed.